index.js

```diff
@@ -81,7 +81,18 @@ function getFlags (cb) {
 // write some json to a file descriptor. if this fails, call back
 // with both the error and the data that was meant to be written.
 function writeConfig (fd, flags, cb) {
-  var buf = new Buffer(JSON.stringify(flags));
+  var json = JSON.stringify(flags);
+  var buf;
+  if (Buffer.from && Buffer.from !== Uint8Array.from) {
+    // Node.js 4.5.0 or newer
+    buf = Buffer.from(json);
+  } else {
+    // Old Node.js versions
+    // The typeof safeguard below is mostly against accidental copy-pasting
+    // and code rewrite, it never happens as json is always a string here.
```

Comment:
If it never happens it's not needed right? Need a test if this is kept in.

Reply:
There were (and are) numerous bugs in the ecosystem caused by such a check missing.

My understanding is that this is not a hot code path so this check doesn't harm (especially given that it's in a branch which targets obsolete and unsupported Node.js versions).

Having this explicit check lowers the probability that some future change (e.g. code restructuring or inheriting) won't cause bugs in that code path.

Reply:
If this was a hot code path, I would have just left a cautious comment there instead.